components/builder-jobsrv/src/server/log.rs

```diff
@@ -0,0 +1,44 @@
+use error::Result;
+use std::fs;
+use std::path::PathBuf;
+
+/// Encapsulates the local filesystem directory in which in-process
+/// build job logs will be collected prior to being sent to long-term
+/// storage.
+#[derive(Clone, Debug)]
+pub struct LogDirectory {
+    path: PathBuf,
+}
+
+impl LogDirectory {
+    /// Create a new `LogDirectory` wrapping `path`.
+    pub fn new(path: &str) -> Self {
+        LogDirectory { path: PathBuf::from(path) }
+    }
+
+    /// Ensures that the specified log directory can be used.
+    ///
+    /// # Panics
+    /// * when the path does not exist
+    /// * when the path is not a directory
+    /// * when the path is not writable
+    pub fn validate(&self) -> Result<()> {
```

Comment:
You would want to return an `Err<T>` where T is `error::Error`. I'd add a descriptive error to `error::Error` which describes that the log directory couldn't be created. I normally wrap the underlying `io::Error` in the case, so I'll define something like `Error::LogDirCreate(PathBuf, io::Error)` and then give it a formatted message like this: `format!("Unable to create log directory at {}, {}", path, err)`